### ISSUE TITLE
Consider adding horizontal scrolling

### ISSUE THREAD

**Issue by eureka-cpu** [helix-editor/helix#8674]
Vertical scrolling with a scroll wheel or track pad seems to already be implemented, but it would be nice to have horizontal scrolling as well.

**Comment by eureka-cpu** [comment]
> _A wild pikachu appeared_
> 
> I think this shouldn't be that hard to implement. I'll claim this!

Feel free to tag me and I'll be happy to test out the feature. Thanks!

**Comment by TornaxO7** [comment]
@eureka-cpu 

May I ask if that's how you imagine it?

Hm... I think it might be better if the horizontal scrolling doesn't use the cursor because you can feel a little "latency" when scrolling your mouse wheel.

https://github.com/helix-editor/helix/assets/50843046/7a240f55-255b-4530-9556-a61b7dade0db

**Comment by eureka-cpu** [comment]
> @eureka-cpu
> 
> May I ask if that's how you imagine it?
> 
> Hm... I think it might be better if the horizontal scrolling doesn't use the cursor because you can feel a little "latency" when scrolling your mouse wheel.
> 

Ideally, the cursor wouldn't move and if you begin typing it would re-focus the cursor 🙂 But I would be happy just to see something like this for now.

**Comment by TornaxO7** [comment]
@eureka-cpu https://github.com/TornaxO7/helix/tree/feat/horizontal-mouse-scrolling

alright! The feature should be implemented now! Could you try it out pls? Basically clone the given repo above and switch to the `feat/horizontal-mouse-scrolling` branch. Afterwards, simply execute `cr -- <path to a file>`.

You can scroll horizontally by holding `shift` and scrolling your mouse wheel.

Uploading Video.mp4…

**Comment by TornaxO7** [comment]
Also for everyone who enabled notifications for this issue: Feel free to try it out :)

**Comment by eureka-cpu** [comment]
I'll give this a try soon, thanks so much for working on this! Hopefully it can be merged soon 🙂

**Comment by eureka-cpu** [comment]
By any chance does this work with trackpad horizontal scrolling as well? Or would you still need to hold shift?

**Comment by TornaxO7** [comment]
> By any chance does this work with trackpad horizontal scrolling as well? Or would you still need to hold shift?

you'd have to hold shift at the moment

**Comment by eureka-cpu** [comment]
@TornaxO7 sorry it's taken me a while to get back to you. Just built and ran it on my M1 and I can't get it to work with the trackpad even with holding shift 🙁

**Comment by TornaxO7** [comment]
> @TornaxO7 sorry it's taken me a while to get back to you. Just built and ran it on my M1 and I can't get it to work with the trackpad even with holding shift 🙁

huh, did you build from the correct branch because it works on my laptop with the trackpad.

**Comment by eureka-cpu** [comment]
Sorry, you were right. I checked out the branch and it looks like there's some strange behaviour. If I scroll down and select somewhere I want to scroll horizontally, it jumps back to the top of the page.

https://github.com/helix-editor/helix/assets/57543709/5b26709e-89fd-43c6-8454-209f82618658

**Comment by TornaxO7** [comment]
@eureka-cpu sorry for the late reply! I haven't forgotten this issue. I hope that this issue isn't urgent because I'd like to fix a terminal issue first xD

**Comment by eureka-cpu** [comment]
> @eureka-cpu sorry for the late reply! I haven't forgotten this issue. I hope that this issue isn't urgent because I'd like to fix a terminal issue first xD

Not at all, I just think it would make the user experience more smooth. No rush 🙂

**Comment by afo** [comment]
@TornaxO7 any update on this? Would love use this feature in helix 🙏

**Comment by TornaxO7** [comment]
No sorry, I'm currently busy by other things so if anyone wants to do this: Feel free to do so.
Maybe I'll come back to this after some time.

![sadge](https://github.com/user-attachments/assets/4b5e1153-f83b-449c-bfbe-b497ad7f7bc7)

**Comment by TornaxO7** [comment]
> [@TornaxO7](https://github.com/TornaxO7) how are things going? Do you think you'd be willing to pick this back up? I'd be happy to take a look at your branch if you update it and take it from there. LMK!

Not for now. I'm currently cooking something else and I'll write some exams in the next weeks but maybe after those.

**Comment by A-kly** [comment]
Any update to this? I would LOVE for this feature to be implemented but i currently have no rust experience!

**Comment by groutoutlook** [comment]
People may feel the utter need of vertical/horizontal scroll bar, but I think `word-wrap` is enough. You guys may consider enabling/toggling word-wrapping in your config.
